presto-hive/src/main/java/com/facebook/presto/hive/PartitionStatistics.java

```diff
@@ -18,70 +18,37 @@
 import com.google.common.collect.ImmutableMap;
 
 import java.util.Map;
-import java.util.OptionalLong;
 
 import static java.util.Objects.requireNonNull;
 
 public class PartitionStatistics
 {
-    public static final PartitionStatistics EMPTY_STATISTICS = new PartitionStatistics(
-            false,
-            OptionalLong.empty(),
-            OptionalLong.empty(),
-            OptionalLong.empty(),
-            OptionalLong.empty(),
-            ImmutableMap.of());
-
-    private final boolean columnStatsAcurate;
-    private final OptionalLong fileCount;
-    private final OptionalLong rowCount;
-    private final OptionalLong rawDataSize;
-    private final OptionalLong totalSize;
+    private final HiveBasicStatistics basicStatistics;
     private final Map<String, HiveColumnStatistics> columnStatistics;
+    private final boolean columnStatsAccurate;
 
     public PartitionStatistics(
-            boolean columnStatsAcurate,
-            OptionalLong fileCount,
-            OptionalLong rowCount,
-            OptionalLong rawDataSize,
-            OptionalLong totalSize,
-            Map<String, HiveColumnStatistics> columnStatistics)
+            HiveBasicStatistics basicStatistics,
+            Map<String, HiveColumnStatistics> columnStatistics,
+            boolean columnStatsAccurate)
```

Comment:
This property is used internally by Hive to execute `SELECT max(column)` kind of queries efficiently. If the statistics are accurate, this query won't scan any data, and will return max value for a give column using statistics.


The thing is that this field is no longer boolean. There is a JSON in it (can't say exactly from what Hive version though). That JSON is a map, that has `boolean basicStatisticsAccurate` and `Map<Column, Boolean> columnStatisticsAccurate` fields. Hive set's those values to `TRUE` after Analyze is run, and it set's those value to `FALSE` if any data modification occurred. 

In Presto we don't do any optimizations of such kind, so we can simply ignore this field (ideally we had to set it to `FALSE` after an INSERT operation, but since we update statistics now - we can just leave it untouched)

Reply:
I think i'm just gonna remove this field from our codebase.